core/commands/commit_compare.py

```diff
@@ -0,0 +1,245 @@
+import sublime
+from sublime_plugin import TextCommand, WindowCommand
+import os
+
+from ..git_command import GitCommand
+from ...common import util
+from ...common import ui
+from .log import GsLogBranchCommand
+
+
+COMMIT_NODE_CHAR = "●"
+COMMIT_NODE_CHAR_OPTIONS = "●*"
+
+
+class GsCompareCommitCommand(TextCommand, GitCommand):
+
+    """
+    Show a view of all commits diff between branches.
+    """
+
+    def run(self, edit, base_commit, target_commit=None, file_path=None, title=None):
+        self.base_commit = base_commit
+        self.target_commit = target_commit if target_commit else "HEAD"
```

Comment:
this does not work since other functions may pass `target_commit=None` to `gs_compare_commit`, you have to make sure that if `target_commit is None` then we set `target_commit` to HEAD.